### PR TITLE
Workflow automatically updates license cache when package-lock.json changes

### PR DESCRIPTION
**Pull request by pje**
This branch is a spike. I'm not convinced it's the right approach, but I wanted to get my thoughts down formally and a Draft PR with working code seemed better than an abstract issue.

### Problem

Whenever we upgrade a dependency, our license cache becomes out of date and the build fails. This is confusing and requires manual intervention for a task that is ultimately Easy To Automate™.

### Approach

The changes in this branch define an Action that automatically regenerates the license caches in response to PRs with changes to `package-lock.json` and commits the result.

### Alternatives

1. A simple pre-commit hook could do the same. Downside: it'd require having a local installation of `licensed`.

### CODE REVIEW

File: .github/workflows/licensed_cache_update.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    runs-on: ubuntu-latest
+    name: Check licenses
+    steps:
+      - uses: actions/checkout@v1
+      - uses: jonabc/setup-licensed@v1.0.2
+        with:
+          version: '2.x'
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+      - run: npm install --production
+      - uses: jonabc/licensed-ci@v1.5.3
```
Comment (thboop) [inline .github/workflows/licensed_cache_update.yml:25]: Consider pinning to v1 unless you have a specific reason to do this

File: .github/workflows/licensed_check.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  pull_request: {branches: main}
+
+jobs:
+  licensed_check:
+    needs: licensed_cache_update
+    runs-on: ubuntu-latest
+    name: Check licenses
+    steps:
+      - uses: actions/checkout@v2
+      - uses: jonabc/setup-licensed@v1.0.2
```
Comment (thboop) [inline .github/workflows/licensed_check.yml:14]: Same as above, consider pinning to v1

File: .github/workflows/licensed_cache_update.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+name: Licensed Cache Update
+
+on:
+  push:
+    branches:
+      - main
+      - licensed_cache_update
```
Comment (thboop) [inline .github/workflows/licensed_cache_update.yml:7]: we should remove this before merging

File: .github/workflows/licensed_cache_update.yml (hunk 1 above)
Comment (pje) [reply .github/workflows/licensed_cache_update.yml:25]: Agree!

**Pull request merged**